### PR TITLE
(Bug) Reserved addresses are not initialized if list length > 20

### LINKED ISSUE #1048: (Bug) Reserved addresses are not initialized if list length  > 20
Steps:
1. Start creating  Minted capped crowdsale, decimals =18, 1 tier, no whitelist
2.  In Step 2:  add 21 reserved addresses from *. csv file 
```
0x66f537cCD03f21c58172602B919884A442A3c313,percentage,4511707483.49603
0x665772109Eb2dc9F5B7c28F987Ec58949d4Eeb87,percentage,5703652631.602385
0x693d436da2c3C11341149522E5F6d0390B363197,tokens,1788191146.7779398
0x67737e0046b05D2553ad1021A02d957D9c854C00,percentage,3813647171.476058
0x20372dffd12d7b012FF44c9DBae70E5e1aE47C73,percentage,6519551480.223114
0x2BCDBBd0Df231e9C35F3a27a4b9F4E631DD6f8AB,percentage,6587800412.130846
0xc4C92D54E2136d7A72F4f738B3FA69Df7b479821,percentage,6802360751.615493
0x2bD96eA633e8BcB468732c68B2CD632BfF4D79Db,tokens,758413595.514329
0x848822D9101ecF8731B09CF52d4a034123FF7F97,percentage,6696003885.57482
0x866623622b19D8603F54d0aF23236759DCeB48d9,tokens,159797903.39822885
0x46fe103Be22C370E2D4143e6b61D1769017b48F0,percentage,7351028916.256665
0x063376DDB05d9e0E0D3761D0D962f9d31F9F6035,percentage,7103593263.293569
0xc6fD167492BeDf4B4f83e84FeCf9483027565650,percentage,2176445015.874138
0x1595F0a0dE18aE149081aA28b2F51aa5619e3523,percentage,8405737163.361933
0xb51454306fAa9A3Cf0505a22dff85d5b4EC13B3F,tokens,1792235377.7930033
0x7A317D210de8201C8623a0f35d44Eb34D8115B45,tokens,6421660777.500122
0x91c93F00F32898269336734Fe0611C1E9CB12d0A,percentage,3777298132.521336
0x61B748ccD326dEF03dE119d74c81391050dFe021,percentage,7330715514.135273
0x24952775eE2c842ea624991a0A422f1B0AcecA84,percentage,4928816077.592806
0xB848714c08acC2D32511C5F93F7BCcd43112F3cb,percentage,7209626039.05291
0x08C9f711d326CaCB341DDBb406BB67475817DAd3,percentage,281316414.4389435
 ```
3. Complete deployment, check if `Register addresses for Reserved Tokens` transaction  is skipped in Step 4

Expected result : 
 -  user should be able to add more than 21 reserved addresses 
OR  warning  'EXCEED THE LIMIT   OF RESERVED ADDRESSES  etc ....  ' should be displayed  in Step 2

Actual result:
 -  Reserved addresses are not initialized if list length  > 20

If you are reporting a problem with Token Wizard, please include the following information:

### Which network did you use? (Mainnet, Kovan, Rinkeby, etc.)
*Ropsten*

### If you were able to create it, what is the URL of your crowdsale?
*n/a*

### Do you have screenshots showing the problem?

screen shot 2018-07-16 at 23 14 16

### Do you see errors in the dev console? If yes, please include a screenshot

---

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #1048 

There is a limitation based on the contracts, see image
screenshot_20180726_141508

When you upload a file that exceeds 20 addresses, a message like this should appear, see image
screenshot_20180726_161613

- Added two new alerts to manage reserved slot address
- Add validation in the ReservedTokenStore
- Refactoring applying new validation

### CODE REVIEW

File: src/utils/alerts.js [hunk 1]
```diff
@@ -278,3 +278,19 @@ export function notAllowedContributor() {
...
+    title: 'No more reserved tokens available',
+    html: `You're not able to reserve more tokens. The maximum allowed is 20`,
+    type: 'info'
+  })
+}
+
+export function noMoreReservedSlotAvailableCSV(count) {
+  return sweetAlert2({
+    title: 'You reach the limit of reserved tokens',
+    html: `You're not able to reserve more tokens. Only ${count} reserved address of the file could be added`,
```
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/utils/alerts.js]: @vbaranov Typo fixed

File: src/stores/ReservedTokenStore.js [hunk 2]
```diff
@@ -52,6 +53,11 @@ class ReservedTokenStore {
       return false
     })
   }
+
+  @computed
+  get validateLength() {
+    return this.tokens.length < 20
```
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/stores/ReservedTokenStore.js]: @vbaranov The constant was added to the helper, and applied

File: src/utils/alerts.js [hunk 3]
```diff
@@ -278,3 +278,19 @@ export function notAllowedContributor() {
     type: 'info'
   })
 }
+
+export function noMoreReservedSlotAvailable() {
+  return sweetAlert2({
+    title: 'No more reserved tokens available',
+    html: `You're not able to reserve more tokens. The maximum allowed is 20`,
```
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/utils/alerts.js]: @vbaranov Constant applied

**Comment by mariano-aguero** [comment]
@vbaranov 
Currently has a validator, which validates when a user adds an address individually, or when a user uploads a CSV file (both actions). It will warning the user with a message, and will list the first 20 addresses added. 
Do you agree with that? Any modification in this process? 
Maybe you prefer not validate when adding addresses, list all the addresses added (even if they are more than 20), and at the moment of pressing the 'Continue' button, validate in that action. Can you confirm this? 
Thanks for the feedback

**Comment by vbaranov** [comment]
>Currently has a validator, which validates when a user adds an address individually, or when a user uploads a CSV file (both actions). It will warning the user with a message, and will list the first 20 addresses added.
Do you agree with that? Any modification in this process?

Yes. I agree. This is what I meant. There is no need to change it. @mariano-aguero could you resolve merging conflicts?

**Comment by mariano-aguero** [comment]
@vbaranov Great, conflict solved!

**Comment by dennis00010011b** [comment]
@mariano-aguero 
Build failed to compile . Error message in console:

```
Failed to compile.

./src/utils/constants.js
  Line 298:  Delete `⏎`  prettier/prettier

Search for the keywords to learn more about each error.
```

**Comment by mariano-aguero** [comment]
@dennis00010011b There was an enter character left, remove it

**Comment by dennis00010011b** [comment]
@vbaranov @mariano-aguero 
Tested it .No issues were found

**Pull request merged**